clients/admin-ui/src/features/common/table/FidesTable.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  ArrowDownIcon,
+  ArrowUpIcon,
+  Flex,
+  Table,
+  TableContainer,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr,
+} from "@fidesui/react";
+import { useRouter } from "next/router";
+import React, { ReactNode } from "react";
+import { Column, useGlobalFilter, useSortBy, useTable } from "react-table";
+
+import GlobalFilter from "~/features/datamap/datamap-table/filters/global-accordion-filter/global-accordion-filter";
+
+interface FidesObject {
+  id?: string;
+  name?: string;
+}
+
+type Props<T extends FidesObject> = {
+  columns: Column<T>[];
+  data: T[];
+  userCanUpdate: boolean;
+  redirectRoute: string;
+  showSearchBar?: boolean;
+  footer?: ReactNode;
+};
+
+export const FidesTable = <T extends FidesObject>({
+  columns,
+  data,
+  userCanUpdate,
+  redirectRoute,
+  showSearchBar,
+  footer,
+}: Props<T>) => {
+  const router = useRouter();
+
+  const tableInstance = useTable({ columns, data }, useGlobalFilter, useSortBy);
+
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
+    tableInstance;
+
+  return (
+    <TableContainer>
+      {showSearchBar ? (
+        <Flex flexGrow={1} marginBottom={3}>
+          <GlobalFilter
```

Comment:
a side note on this global filter: it's great that we can just plug it in right now and it works! but it may not work once we add pagination, since we'd really need to make another backend call and we won't be able to just have the FE search through everything (since it might not have everything!) 

but for now, it's really nice that it just works 😄

Reply:
True. We'll probably need to make it a slot once the search bar needs to hook into the backend